### PR TITLE
[python] Introduce GetClusterInfoById

### LINKED ISSUE #8575: Add mapping or lookup method of Cluster IDs for verification on Test Harness.
#### Problem
We need access to the Cluster IDs present in XML and ZAP files within the test harness. We need this as a source of truth for verification against the cluster IDs we get from DUTs' responses to different cluster commands during testing. 

 #### Proposed Solution
Please add some mapping or lookup capability that we can use to validate the cluster IDs we get from DUT responses against the valid cluster IDs that are predefined by spec and hard coded already within  the xml and zap file. 

Please note that the test harness does not have access to these xml or zap file as we just take in the built code itself.

This issue is blocking (please refer to it for further context) : https://github.com/chip-csg/chip-certification-tool/issues/201

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
What is being fixed?  Examples:
* Add mapping or lookup method of Cluster IDs for verification on Test Harness.
* Fixes #8575

#### Change overview
- Update python/chip/clusters/ChipClusters.py to support looking up method.

#### Testing
- Added `test.TestPythonAPI(self)` to test this API.

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/python/test/test_scripts/base.py [hunk 1]
```diff
@@ -179,3 +196,15 @@ def TestWriteBasicAttributes(self, nodeid: int, endpoint: int, group: int):
             self.logger.exception(f"Following attributes failed: {failed_zcl}")
             return False
         return True
+
+    def TestPythonAPI(self):
```
Comment (trombadore) [inline src/controller/python/test/test_scripts/base.py:200]: please consider changing this def to a more descriptive test name.

`TestPythonAPI` is very generic in this case

**Comment by andy31415** [comment]
Could  you separate out the 'generate file' from the rest of the commits to make code review easier?

**Review by andy31415 — APPROVED** [review]

**Review by trombadore-google — APPROVED** [review]

File: src/controller/python/test/test_scripts/base.py (hunk 1 above)
Comment (erjiaqing) [reply src/controller/python/test/test_scripts/base.py]: I would expect this function will contain more "stateless" API tests for CSG use. Renamed it to TestNonControllerAPIs

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by andy31415** [comment]
@erjiaqing - merge conflicts

**Pull request merged**